models/chunter/src/migration.ts

```diff
@@ -79,8 +86,79 @@ async function createBacklink (tx: TxOperations): Promise<void> {
   }
 }
 
+async function transformCommentsToChatMessages (client: MigrationClient): Promise<void> {
+  while (true) {
+    const comments = await client.find<Comment>(DOMAIN_COMMENT, { _class: chunter.class.Comment }, { limit: 500 })
+
+    if (comments.length === 0) {
+      break
+    }
+
+    for (const comment of comments) {
+      const messageId = generateId<ChatMessage>()
+
+      await client.create<ChatMessage>(DOMAIN_NOTIFICATION, {
+        _id: messageId,
+        _class: notification.class.ChatMessage,
+        space: comment.space,
+
+        attachedTo: comment.attachedTo,
+        attachedToClass: comment.attachedToClass,
+
+        modifiedBy: comment.modifiedBy,
+        modifiedOn: comment.modifiedOn,
+
+        createdBy: comment.createdBy,
+        createdOn: comment.createdOn,
+
+        message: comment.message,
+        isPinned: comment.pinned,
+
+        reactions: comment.reactions,
+        attachments: comment.attachments
+      })
+
+      const attachments = await client.find<Attachment>(DOMAIN_ATTACHMENT, {
+        _class: attachment.class.Attachment,
+        attachedTo: comment._id,
+        attachedToClass: chunter.class.Comment,
+        collection: 'attachments'
+      })
+
+      for (const attach of attachments) {
+        await client.create<Attachment>(DOMAIN_ATTACHMENT, {
+          ...attach,
+          _id: generateId<Attachment>(),
+          attachedTo: messageId,
+          attachedToClass: notification.class.ChatMessage
+        })
+      }
+
+      const reactions = await client.find<Reaction>(DOMAIN_CHUNTER, {
+        _class: chunter.class.Reaction,
+        attachedTo: comment._id,
+        attachedToClass: chunter.class.Comment,
+        collection: 'reactions'
+      })
+
+      for (const reaction of reactions) {
+        await client.create<Reaction>(DOMAIN_CHUNTER, {
+          ...reaction,
+          _id: generateId<Reaction>(),
+          attachedTo: messageId,
+          attachedToClass: notification.class.ChatMessage
+        })
+      }
+
+      await client.delete(DOMAIN_COMMENT, comment._id)
```

Comment:
You may delete by bulk

Reply:
I'm not sure I understood correctly
Used `deleteMany`